data/api_route_data_test.go

```diff
@@ -0,0 +1,296 @@
+	return request, response
+}
+
+// Utility function to get default params passed to store.GetDeviceData
+func getDataStoreDefaultParams() store.Params {
+	return store.Params{
+		UserID:        "patient",
+		Types:         []string{""},
+		SubTypes:      []string{""},
+		SchemaVersion: &schemaVersions,
+		LevelFilter:   []int{0, 1},
+		Carelink:      true,
+		Medtronic:     true,
+	}
+}
+
+// TestGetData_NoToken calling GetData route without token should be unauthorized
+func TestGetData_NoToken(t *testing.T) {
+	resetMocks()
+
+	urlParams := make(map[string]string)
+	request, response := getDataPrepareRequest("", urlParams)
+	tidewhisperer.GetData(response, request, defaultGetDataURLVars)
+
+	checkUnAuthorized(response, t)
+}
+
+// TestGetData_WrongToken calling GetData route with an authorized token should be unauthorized
+func TestGetData_WrongToken(t *testing.T) {
+	resetMocks()
+	mockShoreline.Unauthorized = true
+
+	urlParams := make(map[string]string)
+	request, response := getDataPrepareRequest("mytoken", urlParams)
+	tidewhisperer.GetData(response, request, defaultGetDataURLVars)
+
+	checkUnAuthorized(response, t)
+}
+
+// TestGetData_GoodTokenSameUser calling GetData route for the user owning the token should be authorized
+func TestGetData_GoodTokenSameUser(t *testing.T) {
+	resetMocks()
+
+	urlParams := make(map[string]string)
+	request, response := getDataPrepareRequest("mytoken", urlParams)
+	tidewhisperer.GetData(response, request, defaultGetDataURLVars)
+
+	checkAuthorized(response, t)
+	getDataCheckStoreParams(defaultGetDataStoreParams, t)
+}
+
+// TestGetData_GoodTokenGuestUserNotInvited calling GetData route for a user who didn't invite the user owning the token should be unauthorized
+func TestGetData_GoodTokenGuestUserNotInvited(t *testing.T) {
+	resetMocks()
+	mockShoreline.UserID = "guestUninvited"
+	mockPerms.SetExpected(clients.Permissions{"unauthorized": clients.Permission{}}, nil)
+
+	urlParams := make(map[string]string)
+	request, response := getDataPrepareRequest("mytoken", urlParams)
+	tidewhisperer.GetData(response, request, defaultGetDataURLVars)
+
+	checkUnAuthorized(response, t)
+}
+
+// TestGetData_GoodTokenGuestUserNotInvited calling GetData route for a user who invited the user owning the token should be authorized
+func TestGetData_GoodTokenGuestUserInvited(t *testing.T) {
+	resetMocks()
+	mockShoreline.UserID = "guest"
+
+	urlParams := make(map[string]string)
+	request, response := getDataPrepareRequest("mytoken", urlParams)
+	tidewhisperer.GetData(response, request, defaultGetDataURLVars)
+
+	checkAuthorized(response, t)
+	getDataCheckStoreParams(defaultGetDataStoreParams, t)
+}
+
+// TestGetData_ServerToken calling GetData route for any user with a server token should be authorized
+func TestGetData_ServerToken(t *testing.T) {
+	resetMocks()
+	mockShoreline.UserID = "server"
+	mockShoreline.IsServer = true
+	mockPerms.SetExpected(clients.Permissions{"unauthorized": clients.Permission{}}, nil)
```

Comment:
This test is about access to the route with a server token. 
With a server token we should always access data even if gatekeeper don't authorize it. In fact gatekeeper is never called with a server token, the pretty way to test it would be to assert that the mocked geatekeeper method hasn't been called... But we don't have a "spy" on that mock.

I'm OK that this line is useless, it was just to be more explicit on what we wanted to test.